x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomLink/CustomLinkFlyout/DeleteButton.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonEmpty } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { NotificationsStart } from 'kibana/public';
+import React, { useState } from 'react';
+import { useCallApmApi } from '../../../../../../hooks/useCallApmApi';
+import { APMClient } from '../../../../../../services/rest/createCallApmApi';
+import { useApmPluginContext } from '../../../../../../hooks/useApmPluginContext';
+
+interface Props {
+  onDelete: () => void;
+  customLinkId: string;
+}
+
+export function DeleteButton({ onDelete, customLinkId }: Props) {
+  const [isDeleting, setIsDeleting] = useState(false);
+  const { toasts } = useApmPluginContext().core.notifications;
+  const callApmApi = useCallApmApi();
```

Comment:
Am I the only one who dies a little bit every time I see us calling this hook, so we can pass it as an argument into another function. 

I know, I know. Can't simply import it where we need it. Too easy! Might be stateful so needs to be passed down meticulously through the entire app. Ugh :(

My dirty idea: for `callApmApi` specifically (not all the other `core` deps), what about exposing that as a singleton that can be imported like:

```js
import { callApmApi } from 'apm/api'
```

That almost looks too good to be legal, doesn't it? 🤩

Reply:
@sqren https://github.com/elastic/kibana/pull/57788/commits/3f14e87d8c7439866566c2e4f379d69c09302893 LMK!

Reply:
I like it!

Reply:
Looks good :+1:

Little bit of NP trivia, this was the pattern Court suggested for sharing Core inside deeply nested components before teams started adopting the ContextProvider + hooks pattern.